### PR TITLE
doc: basic tutorial

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Introductory tutorial 

```cmd
make serve-doc
```

image

image

## Checklist

- [x] ``test/test_workflows.py`` runs against fake server
- [x] Is [``api/openapi.json``](./api/openapi.json) up-to-date
- [x] Auto-doc generated 
- [x] Links to notebooks modified

### CODE REVIEW

**Comment by pcrespov** [comment]
@KZzizzle this is a first draft of the introductory tutorial we discussed about last Friday. Please have a first look at it and we can iterate over it a couple of times more.

File: docs/md/BasicTutorial.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+```command
+$ pip install osparc
+$ python -c "import osparc; print(osparc.__version__)"
+```
+
+
+## Setup
+
+To setup the client, we need to provide a username and password to the configuration. These can be obtained in the UI under [Preferences > API Settings > API Keys](http://docs.osparc.io/#/docs/platform_introduction/main_window_and_navigation/user_setup/profile?id=preferences). Use the *API key* as username and the *API secret* as password. For security reasons, you should not write these values in your script but instead set them up via environment variables or read them from a separate file. In this example, we use environment variables
```
Comment (KZzizzle) [inline docs/md/BasicTutorial.md:16]: just add "which will be referred to as "OSPARC_API_KEY" and "OSPARC_API_SECRET" for the rest of the tutorial."

File: docs/md/BasicTutorial.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    #  'role': 'USER'}
+    #
+```
+
+
+## Solvers Workflow
+
+The osparc API can be used to execute any computational service published in the platform. This means that any computational service listed in the UI under the [Discover Tab](http://docs.osparc.io/#/docs/platform_introduction/core_elements/Discover?id=discover-tab) is accessible from the API. Note that computational services are denoted as *solvers* in the API for convenience, but they refer to the same concept.
+
+Let's use the sleepers computational service to illustrate a typical workflow.
```
Comment (KZzizzle) [inline docs/md/BasicTutorial.md:71]: Could we add "The sleepers computational service takes as input one natural number and an optional text file input that contains another natural number, and then sleeps for the designated number of seconds based on inputs. It also provides two outputs: one natural number and a file containing a single natural number."

File: docs/md/BasicTutorial.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+```
+
+In the snippet above, ``input_file`` is a ``File`` reference to the uploaded file and that is passed as input to the solver. Analogously, ``results_file`` is a ``File`` produced by the solver and that can also be downloaded.
+
+
+#### Solvers, Inputs and Outputs
+
+The inputs and outputs are specific for every solver. Every input/output has a name and an associated type that can be as simple as booleans, numbers, strings ... or more complex as files. You can find this information in the UI under Discover Tab, selecting the service card > More Info > raw metadata. For instance, the ``sleeper`` version ``2.1.1`` has the following ``raw-metadata``:
+
+```json
```
Comment (KZzizzle) [inline docs/md/BasicTutorial.md:172]: I'm not sure this version of the service is available on osparc production?

**Review by KZzizzle — COMMENTED** [review]
Very cool! I made a few comments but overall it's great!

File: docs/md/BasicTutorial.md (hunk 3 above)
Comment (pcrespov) [reply docs/md/BasicTutorial.md]: true. I adapted it to 2.0.2.

**Pull request merged**